### DISCUSSION TITLE
Override the CDP

### DISCUSSION THREAD

**Discussion by htdangkhoa** [browserless/chrome#4275] [Q&A] [answered]
Hello everyone, I'm a developer using Puppeteer and I'm interested in you guys customizing CDP's methods and events, I see you've already done this. I asked the Chromium team a question and received the answer that customizing CDP requires rebuilding the browser from the Chromium source code. Can you share how you customized CDP? Do you have to rebuild the browser?

I appreciate all of your opinions, thank you very much!

**Comment by joelgriffith** [comment, marked as answer]
We didn't build Chromium from source, but instead have a websocket server act as a proxy to Chrome. Anytime one of our "Browserless" events comes through we manually handle that, otherwise we allow the rest of the actions to proceed. Let me know if that's helpful or makes sense?

**Reply by htdangkhoa** [reply]
I'm clear, thank you for your answer
